virtual_ecosystem/models/soil/soil_model.py

```diff
@@ -124,6 +130,8 @@ def __init__(
             or np.any(data["soil_c_pool_necromass"] < 0.0)
             or np.any(data["soil_n_pool_don"] < 0.0)
             or np.any(data["soil_n_pool_particulate"] < 0.0)
+            or np.any(data["soil_n_pool_necromass"] < 0.0)
```

Comment:
Considering how big this condition is, I wonder if it would be better to make it a separate function. Just for readability purposes. Something like:

```python
def _all_pools_positive(data: Data) -> bool:
    """Checks if all pools have zero or positive values.
    return ...
```

So this becomes:

```python
if not _all_pools_positive(data):
    raise ...
```